content/tutorials/content-types/upload-content-with-a-custom-type.md

```diff
@@ -0,0 +1,115 @@
+
+Then *Image* becomes available in the add new content dropdown and if you upload an image to the library it will be created with the `Image` type.
+
+![Upload image as File](./img/upload_as_image.gif "Upload image as File")
+
+## Upload content with a custom type
+
+Let's create a custom type inherited from File with some additional fields for Contracts. Create a new content type definition with the following xml:
+
+```xml
+<ContentType name="Contract" parentType="File" handler="SenseNet.ContentRepository.File" xmlns="http://schemas.sensenet.com/SenseNet/ContentRepository/ContentTypeDefinition">
+  <DisplayName>Contract</DisplayName>
+  <Description>Type for agreement documents</Description>
+  <Icon>TaskList</Icon>
+  <Fields>
+  <Field name="ContractId" type="ShortText">
+    <DisplayName>Contract Id</DisplayName>
+    <Description>Unique contract Id needed for example for legal proceedings</Description>
+    <Configuration>
+      <MaxLength>30</MaxLength>
+      <MinLength>10</MinLength>
+      <Regex>[a-zA-Z0-9]*$</Regex>
+      <Compulsory>true</Compulsory>
+    </Configuration>
+  </Field>
+  <Field name="RelatedProject" type="Reference">
+    <DisplayName>Related project</DisplayName>
+    <Description>Project to which the contract relates</Description>
+    <Configuration>
+      <AllowMultiple>false</AllowMultiple>
+      <AllowedTypes>
+        <Type>ProjectWorkspace</Type>
+      </AllowedTypes>
+    </Configuration>
+  </Field>
+  <Field name="ContractValue" type="Number">
+    <DisplayName>Contract value</DisplayName>
+    <Description>Value of the contract</Description>
+    <Configuration>
+      <Digits>2</Digits>
+    </Configuration>
+  </Field>
+  <Field name="RenewalDate" type="DateTime">
+    <DisplayName>Renewal date</DisplayName>
+    <Description>The date of renewal of the contract</Description>
+  </Field>
+  </Fields>
+</ContentType>
+```
+
+This content type definition will include fields inherited from the File type (*Binary*, *DisplayName*, etc) and custom ones related to contracts as mentioned in the example above (*RelatedProject*, *RenewalDate*, etc).
+
+Open the settings menu and go to Configurations. There you can find a configurations file names Portal.settings. Open it for edit. This json file contains system-wide settings related to extensions and file types. Search for the list of `UploadFileExtensions`, which is actually an array that contains extension and content type pairs. You can define here how the system should handle the different file extensions when you upload a file with a specific one. As you can see, the common image file extensions are listed and connected to the *Image* type and *File* is the default. Add a new key value pair to connect Word documents and your newly created type, *Contract*. This setting means that if you upload a Word document it will be created as Contract and will have the custom fields as well.
```

Comment:
```suggestion
Open the settings menu and go to Configurations. There you can find a configurations file named Portal.settings. Open it for edit. This json file contains system-wide settings related to extensions and file types. Search for the list of `UploadFileExtensions`, which is actually an array that contains extension and content type pairs. You can define here how the system should handle the different file extensions when you upload a file with a specific one. As you can see, the common image file extensions are listed and connected to the *Image* type and *File* is the default. Add a new key value pair to connect Word documents and your newly created type, *Contract*. This setting means that if you upload a Word document it will be created as Contract and will have the custom fields as well.
```